ocaml/testsuite/tests/typing-modes/stack.ml

```diff
@@ -0,0 +1,103 @@
+(* TEST
+expect;
+*)
+
+let f = ref (stack_ (fun x -> x))
+[%%expect{|
+Line 1, characters 20-32:
+1 | let f = ref (stack_ (fun x -> x))
```

Comment:
Not sure what you mean - surely this `fun x -> x` is an allocation? Or do you mean it's hoisted?

Reply:
Like, should we expose to the user the fact that `fun x -> x` is not an allocation because of optimizations? I think not.

Also, to detect this in the type checker is difficult - will need to defer to middle-end. But middle-end printing error messages sounds bad.

Reply:
Yes, good point. The constant-check is in transl, but I misremembered thinking it was in the type-checker. And the check I’m thinking of is conservative, because it happens before inlining. So I’ll just resolve.